### PR TITLE
Add and fix several Cisco device types

### PR DESCRIPTION
**Pull request by desnoe**

### CODE REVIEW

**Comment by danner26** [comment]
Please fix all tests @desnoe and then ping me once they are all passing. In the meantime I will revert this PR to a draft state

**Comment by desnoe** [comment]
@danner26 all tests are now passing

**Comment by danner26** [comment]
Hello, unfortunately today we had to make a change to the schema in which weight is defined by. The upstream NetBox API requires weight to be passed to it in the following format:
```
weight: 12.01
weight_unit: lb
```

Due to this change, I have updated your branch against the current master. This might require you to update your PR, if you had weight currently defined, so that your PR is in-line with the new weight schema. If you have any issues, please let me know or refer to [this PR](https://github.com/netbox-community/devicetype-library/pull/1187) for more information.

**Comment by desnoe** [comment]
No problem @danner26 it seems to be passing all tests.

**Comment by danner26** [comment]
Hey @desnoe all devices from now on should have their manufacturer name appended to their slugs. I.E. `cisco-` in this case. Can you please make sure that modification is made? Thank you

**Comment by desnoe** [comment]
> Hey @desnoe all devices from now on should have their manufacturer name appended to their slugs. I.E. `cisco-` in this case. Can you please make sure that modification is made? Thank you

This is now fixed.

File: device-types/Cisco/WS-C2960XR-24TD-I.yaml [hunk 1]
```diff
@@ -75,5 +75,7 @@ console-ports:
 module-bays:
   - name: PSU0
     position: '0'
+    type: iec-60320-c14
```
Comment (danner26) [inline device-types/Cisco/WS-C2960XR-24TD-I.yaml:78]: You dont define PSU type in the module bay, that is part of the module-type file

File: device-types/Cisco/WS-C2960XR-24TD-I.yaml [hunk 2]
```diff
@@ -75,5 +75,7 @@ console-ports:
...
   - name: PSU1
     position: '1'
+    type: iec-60320-c14
```
Comment (danner26) [inline device-types/Cisco/WS-C2960XR-24TD-I.yaml:81]: You dont define PSU type in the module bay, that is part of the module-type file

File: device-types/Cisco/WS-C2950G-24-EI.yaml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+slug: cisco-ws-c2950g-24-ei
+u_height: 1
+is_full_depth: false
+console-ports:
+  - name: con 0
+    type: rj-45
+power-ports:
+  - name: POWER
+    type: iec-60320-c14
+    maximum_draw: 45
```
Comment (danner26) [inline device-types/Cisco/WS-C2950G-24-EI.yaml:14]: Dont these have a DC secondary power supply port? Specifically for a 14 pin DC terminal? If so, please add a power port with `dc-terminal` type

File: device-types/Cisco/WS-C2950G-24-EI.yaml [hunk 4]
```diff
@@ -0,0 +1,67 @@
+---
+manufacturer: Cisco
+model: Catalyst 2950G-24-EI
+part_number: WS-C2950G-24-EI
+slug: cisco-ws-c2950g-24-ei
+u_height: 1
+is_full_depth: false
```
Comment (danner26) [inline device-types/Cisco/WS-C2950G-24-EI.yaml:7]: Do you have `weight`? If so, please add it

File: device-types/Cisco/WS-C2950G-24-EI-DC.yaml [hunk 5]
```diff
@@ -0,0 +1,63 @@
+---
+manufacturer: Cisco
+model: Catalyst 2950G-24-EI-DC
+part_number: WS-C2950G-24-EI-DC
+slug: cisco-ws-c2950g-24-ei-dc
+u_height: 1
+is_full_depth: false
```
Comment (danner26) [inline device-types/Cisco/WS-C2950G-24-EI-DC.yaml:7]: Do you have `weight`? If so, please add it

File: device-types/Cisco/NCS-57B1-5DSE-SYS.yaml [hunk 6]
```diff
@@ -0,0 +1,81 @@
+---
+manufacturer: Cisco
+model: NCS 57B1-5DSE
+slug: cisco-ncs-57b1-5dse-sys
+part_number: NCS-57B1-5DSE-SYS
+u_height: 1
+is_full_depth: true
```
Comment (danner26) [inline device-types/Cisco/NCS-57B1-5DSE-SYS.yaml:7]: Do you have `weight`? If so, please add it

File: device-types/Cisco/N3K-C3548P-10G.yaml [hunk 7]
```diff
@@ -0,0 +1,122 @@
+---
+manufacturer: Cisco
+model: Nexus 3548P-10G
+slug: cisco-n3k-c3548p-10g
+part_number: N3K-C3548P-10G
+u_height: 1
+is_full_depth: false
```
Comment (danner26) [inline device-types/Cisco/N3K-C3548P-10G.yaml:7]: Do you have `weight`? If so, please add it

File: device-types/Cisco/ME-3400EG-12CS-M.yaml [hunk 8]
```diff
@@ -0,0 +1,46 @@
+---
+manufacturer: Cisco
+model: ME 3400EG-12CS-M
+slug: cisco-me-3400eg-12cs-m
+part_number: ME-3400EG-12CS-M
+is_full_depth: false
+u_height: 1
```
Comment (danner26) [inline device-types/Cisco/ME-3400EG-12CS-M.yaml:7]: Do you have `weight`? If so, please add it

File: device-types/Cisco/ASAv.yaml [hunk 9]
```diff
@@ -0,0 +1,7 @@
+---
+manufacturer: Cisco
+model: ASAv
+slug: cisco-asav
+part_number: ASAv
+is_full_depth: false
+u_height: 0
```
Comment (danner26) [inline device-types/Cisco/ASAv.yaml:7]: Do you have `weight`? If so, please add it

Comment (danner26) [inline device-types/Cisco/ASAv.yaml:7]: Does this device have any ports? Or was this just not finished yet?

File: device-types/Cisco/ASA5515.yaml [hunk 10]
```diff
@@ -0,0 +1,30 @@
+---
+manufacturer: Cisco
+model: ASA5515
+slug: cisco-asa5515
+part_number: ASA5515
+is_full_depth: false
+u_height: 1
```
Comment (danner26) [inline device-types/Cisco/ASA5515.yaml:7]: Do you have `weight`? If so, please add it

File: device-types/Cisco/WS-C2970G-24TS-E.yaml [hunk 11]
```diff
@@ -0,0 +1,70 @@
+---
+manufacturer: Cisco
+model: Catalyst 2970G-24TS-E
+part_number: WS-C2970G-24TS-E
+slug: cisco-ws-c2970g-24ts-e
+u_height: 1
+is_full_depth: false
```
Comment (danner26) [inline device-types/Cisco/WS-C2970G-24TS-E.yaml:7]: Do you have `weight`? If so, please add it

File: device-types/Cisco/WS-C3550-12T.yaml [hunk 12]
```diff
@@ -0,0 +1,38 @@
+---
+manufacturer: Cisco
+model: Catalyst 3550-12T
+part_number: WS-C3550-12T
+slug: cisco-ws-c3550-12t
+u_height: 1
+is_full_depth: false
```
Comment (danner26) [inline device-types/Cisco/WS-C3550-12T.yaml:7]: Do you have `weight`? If so, please add it

File: device-types/Cisco/WS-C3560G-24TS-E.yaml [hunk 13]
```diff
@@ -0,0 +1,74 @@
+---
+manufacturer: Cisco
+model: Catalyst 3560G-24TS-E
+slug: cisco-ws-c3560g-24ts-e
+part_number: WS-C3560G-24TS-E
+is_full_depth: false
+u_height: 1
```
Comment (danner26) [inline device-types/Cisco/WS-C3560G-24TS-E.yaml:7]: Do you have `weight`? If so, please add it

File: device-types/Cisco/WS-C3560G-48TS-E.yaml [hunk 14]
```diff
@@ -0,0 +1,117 @@
+---
+manufacturer: Cisco
+model: Catalyst 3560G-48TS-E
+slug: cisco-ws-c3560g-48ts-e
+part_number: WS-C3560G-48TS-E
+u_height: 1
```
Comment (danner26) [inline device-types/Cisco/WS-C3560G-48TS-E.yaml:6]: Do you have `weight`? If so, please add it

File: device-types/Cisco/WS-C3560V2-48TS-E.yaml [hunk 15]
```diff
@@ -0,0 +1,117 @@
+---
+manufacturer: Cisco
+model: Catalyst 3560V2-48TS-E
+slug: cisco-ws-c3560v2-48ts-e
+part_number: WS-C3560V2-48TS-E
+u_height: 1
```
Comment (danner26) [inline device-types/Cisco/WS-C3560V2-48TS-E.yaml:6]: Do you have `weight`? If so, please add it

File: device-types/Cisco/WS-C3560X-24T-L.yaml [hunk 16]
```diff
@@ -0,0 +1,67 @@
+---
+manufacturer: Cisco
+model: Catalyst 3560X-24T-L
+slug: cisco-ws-c3560x-24t-l
+part_number: WS-C3560X-24T-L
+is_full_depth: false
+u_height: 1
```
Comment (danner26) [inline device-types/Cisco/WS-C3560X-24T-L.yaml:7]: Do you have `weight`? If so, please add it

File: device-types/Cisco/WS-C3750X-24S-S.yaml [hunk 17]
```diff
@@ -0,0 +1,73 @@
+---
+manufacturer: Cisco
+model: Catalyst 3750X-24S-S
+slug: cisco-ws-c3750x-24s-s
+part_number: WS-C3750X-24S-S
+is_full_depth: false
+u_height: 1
```
Comment (danner26) [inline device-types/Cisco/WS-C3750X-24S-S.yaml:7]: Do you have `weight`? If so, please add it

File: device-types/Cisco/WS-C4510R.yaml [hunk 18]
```diff
@@ -0,0 +1,22 @@
+---
+manufacturer: Cisco
+model: Catalyst C4510R
+slug: cisco-ws-c4510r
+part_number: WS-C4510R
+u_height: 14
+is_full_depth: true
```
Comment (danner26) [inline device-types/Cisco/WS-C4510R.yaml:7]: Do you have `weight`? If so, please add it

Comment (danner26) [inline device-types/Cisco/WS-C4510R.yaml:7]: Do you have `airflow`? If so, please add it

File: device-types/Cisco/WS-C3750X-24S-S.yaml (hunk 17 above)
Comment (danner26) [inline device-types/Cisco/WS-C3750X-24S-S.yaml:7]: Do you have `airflow`? If so, please add it

File: device-types/Cisco/WS-C3560X-24T-L.yaml (hunk 16 above)
Comment (danner26) [inline device-types/Cisco/WS-C3560X-24T-L.yaml:7]: Do you have `airflow`? If so, please add it

File: device-types/Cisco/WS-C3560V2-48TS-E.yaml (hunk 15 above)
Comment (danner26) [inline device-types/Cisco/WS-C3560V2-48TS-E.yaml:6]: Do you have `airflow`? If so, please add it

File: device-types/Cisco/WS-C3560G-48TS-E.yaml (hunk 14 above)
Comment (danner26) [inline device-types/Cisco/WS-C3560G-48TS-E.yaml:6]: Do you have `airflow`? If so, please add it

File: device-types/Cisco/WS-C3560G-24TS-E.yaml (hunk 13 above)
Comment (danner26) [inline device-types/Cisco/WS-C3560G-24TS-E.yaml:7]: Do you have `airflow`? If so, please add it

File: device-types/Cisco/WS-C3550-12T.yaml (hunk 12 above)
Comment (danner26) [inline device-types/Cisco/WS-C3550-12T.yaml:7]: Do you have `airflow`? If so, please add it

File: device-types/Cisco/WS-C2970G-24TS-E.yaml (hunk 11 above)
Comment (danner26) [inline device-types/Cisco/WS-C2970G-24TS-E.yaml:7]: Do you have `airflow`? If so, please add it

File: device-types/Cisco/WS-C2950G-24-EI.yaml (hunk 4 above)
Comment (danner26) [inline device-types/Cisco/WS-C2950G-24-EI.yaml:7]: Do you have `airflow`? If so, please add it

File: device-types/Cisco/WS-C2950G-24-EI-DC.yaml (hunk 5 above)
Comment (danner26) [inline device-types/Cisco/WS-C2950G-24-EI-DC.yaml:7]: Do you have `airflow`? If so, please add it

File: device-types/Cisco/NCS-57B1-5DSE-SYS.yaml (hunk 6 above)
Comment (danner26) [inline device-types/Cisco/NCS-57B1-5DSE-SYS.yaml:7]: Do you have `airflow`? If so, please add it

File: device-types/Cisco/N3K-C3548P-10G.yaml (hunk 7 above)
Comment (danner26) [inline device-types/Cisco/N3K-C3548P-10G.yaml:7]: Do you have `airflow`? If so, please add it

File: device-types/Cisco/ME-3400EG-12CS-M.yaml (hunk 8 above)
Comment (danner26) [inline device-types/Cisco/ME-3400EG-12CS-M.yaml:7]: Do you have `airflow`? If so, please add it

File: device-types/Cisco/ASAv.yaml (hunk 9 above)
Comment (danner26) [inline device-types/Cisco/ASAv.yaml:7]: Do you have `airflow`? If so, please add it

File: device-types/Cisco/ASA5515.yaml (hunk 10 above)
Comment (danner26) [inline device-types/Cisco/ASA5515.yaml:7]: Do you have `airflow`? If so, please add it

**Review by danner26 — CHANGES_REQUESTED** [review]
Here are my initial comments, please notate if you dont have information requested and then you can resolve it.. Otherwise please add the applicable information then resolve it. Thanks!

**Pull request closed without merging**